### PR TITLE
Replacing nosetests with pytest

### PR DESCRIPTION
**Pull request by mathurinm**

### CODE REVIEW

File: doc/auto_examples/plot_leukemia_path.py [hunk 1]
```diff
@@ -1,65 +0,0 @@
...
-
-    t0 = time.time()
-    _, coefs, dual_gaps = lasso_path(X, y, tol=tol, alphas=alphas)
-    results[1, tol_ix] = time.time() - t0
-    coefs = coefs.T
-
-labels = [r"\sc{Celer}", "scikit-learn"]
-figsize = (7, 4)
-fig = plot_path_hist(results, labels, tols, figsize, ylim=None)
-plt.show()
```
Comment (agramfort) [inline doc/auto_examples/plot_leukemia_path.py]: why removing this example?

Comment (mathurinm) [reply doc/auto_examples/plot_leukemia_path.py]: it's the one in auto_examples (not examples/) that you committed unknowingly a while ago

**Comment by agramfort** [comment]
oups...

**Comment by mathurinm** [comment]
@agramfort 5 tests pass locally, I get one warning:
```
celer/tests/test_homotopy.py::test_LassoCV_compatibility
  /home/mathurin/anaconda3/lib/python3.5/site-packages/sklearn/linear_model/coordinate_descent.py:491: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
    ConvergenceWarning)
```
I cannot ignore it in setup.cfg, it gives
`INTERNALERROR> warnings._OptionError: unknown warning category: 'ConvergenceWarning'`


Does the warning cause the build failure?

**Comment by agramfort** [comment]
Why do you want to ignore it via setup config ? You can capture it in test or live with it

**Comment by mathurinm** [comment]
OK, I thought it caused the build to crash. Any idea of the culprit if it is not the warning?

**Comment by mathurinm** [comment]
ok maybe the `pytest -m`

**Comment by mathurinm** [comment]
@agramfort  Should I remove the failing test case (fit intercept in sparse case, it fails on master also), merge this and address the issue in another PR?

**Comment by agramfort** [comment]
Yes

**Pull request merged**